books/serializers.py

```diff
@@ -0,0 +1,59 @@
+from rest_framework import serializers
+
+from books.models.books import Book, Author, Publisher
+from books.models.country import Country
+
+
+class BookSerializer(serializers.ModelSerializer):
+    id = serializers.IntegerField(read_only=True)
+    publisher = serializers.SlugRelatedField(slug_field='name', read_only=True)
+    country = serializers.SlugRelatedField(slug_field='name', read_only=True)
+    authors = serializers.SlugRelatedField(slug_field='name', many=True, read_only=True)
+
+    class Meta:
+        model = Book
+        fields = ("id", "name", "isbn", "authors", "number_of_pages", "publisher", "country", "release_date")
+
+    def to_representation(self, instance):
+        data = super(BookSerializer, self).to_representation(instance)
+
+        return data
+
+    def create(self, validated_data):
+        request = self.context['request']
+        country, _ = Country.objects.get_or_create(name=request.data.get('country'))
+        publisher, _ = Publisher.objects.get_or_create(name=request.data.get('publisher'))
+        instance = Book.objects.create(country=country, publisher=publisher, **validated_data)
+        authors = request.data.get('authors')
+        for author in authors:
+            author, status = Author.objects.get_or_create(name=author)
+            instance.authors.add(author)
+        return instance
+
+    def update(self, instance, validated_data):
+        request = self.context['request']
+        instance.name = validated_data.get('name', instance.name)
```

Comment:
@xhahid43eb Why you need to manually assign these values here, I think those should be assigned to book instance when you instantiate `Bookserializer` with book object, wasn't that working for you?

Reply:
I do this because updating the relation data need to be assign manually